validator/server_arb/machine.go

```diff
@@ -112,6 +115,14 @@ func (m *ArbitratorMachine) Clone() *ArbitratorMachine {
 	defer m.mutex.Unlock()
 	newMach := machineFromPointer(C.arbitrator_clone_machine(m.ptr))
 	newMach.contextId = m.contextId
+
+	if m.contextId != nil {
+		resolverWithRefCounter, ok := preimageResolvers.Load(*m.contextId)
```

Comment:
Same thing here about logging an error if it's not found

Reply:
This log will only occur if there is a bug in Nitro, right?

For example, it could happen if Clone is called after a machine is destroyed, but in that case I suppose that C.arbitrator_clone_machine call will "panic" before this log, since m.ptr will be nil.

In scenarios similar to this one I prefer to panic instead of only logging.
I consider that Nitro will be a inconsistent state which I would not consider reliable anymore, so I prefer to stop execution, and restart it in a consistent state.

This is also true for the log proposed here https://github.com/OffchainLabs/nitro/pull/2752#discussion_r1826454918

LMK what you think.